chia/full_node/mempool_manager.py

```diff
@@ -491,13 +519,25 @@ async def validate_spend_bundle(
                 return tl_error, None, []  # MempoolInclusionStatus.FAILED
 
         if fail_reason is Err.MEMPOOL_CONFLICT:
-            for conflicting in conflicts:
-                for c_sb_id in self.mempool.removals[conflicting.name()]:
-                    sb: MempoolItem = self.mempool.spends[c_sb_id]
-                    conflicting_pool_items[sb.name] = sb
-            log.warning(f"Conflicting pool items: {len(conflicting_pool_items)}")
-            if not self.can_replace(conflicting_pool_items, removal_record_dict, fees, fees_per_cost):
-                return Err.MEMPOOL_CONFLICT, potential, []
+            conflicts = []
+            for coin in potential_conflicts:
+                # See if this coin spend is eligible for deduplication
+                coin_id = coin.name()
+                eligible_coin_spends_in_item = [
```

Comment:
I'm not sure exactly what you're doing here. you need to know whether a coin spend is eligible or not. You can just build a set of all eligible coins up-front, before looping over `potential_conflicts`.

Right now, the mempool has an invariant that no coin is spent multiple times, right? This is a big change, and should have some comments added in a few places to make this clear.